src/main/java/org/nuxeo/wopi/jaxrs/FilesEndpoint.java

```diff
@@ -0,0 +1,613 @@
+     */
+    protected Response buildConflictResponse(String currentLock) {
+        // locked by another WOPI client
+        response.addHeader(LOCK, currentLock);
+        return Response.status(Response.Status.CONFLICT).build();
+    }
+
+    /**
+     * Implements the GetLock operation.
+     * <p>
+     * See <a href="https://wopi.readthedocs.io/projects/wopirest/en/latest/files/GetLock.html"></a>.
+     */
+    protected Object getLock() {
+        if (!doc.isLocked()) {
+            response.addHeader(LOCK, "");
+            return Response.ok().build();
+        }
+
+        String currentLock = getCurrentLock();
+        response.addHeader(LOCK, currentLock);
+        return Response.ok().build();
+    }
+
+    protected Object unlockOrRefresh(String lock, boolean unlock) {
+        boolean isLocked = doc.isLocked();
+        if (!isLocked) {
+            // not locked
+            response.addHeader(LOCK, "");
+            throw new ConflictException();
+        }
+
+        String currentLock = getCurrentLock();
+        if (lock.equals(currentLock)) {
+            checkWritePropertiesPermission();
+            if (unlock) {
+                // unlock
+                doc.removeLock();
+                store.put(doc.getId(), (String) null);
+                response.addHeader(ITEM_VERSION, doc.getVersionLabel());
+            } else {
+                // refresh lock
+                store.setTTL(doc.getId(), 30 * 60 * 1000);
+            }
+            return Response.ok().build();
+        }
+
+        return buildConflictResponse(currentLock);
+    }
+
+    /**
+     * Implements the PutRelativeFile operation.
+     * <p>
+     * We do not handle any conflict or overwrite here. Nuxeo can have more than one document with the same title and
+     * blob file name.
+     * <p>
+     * See <a href="https://wopi.readthedocs.io/projects/wopirest/en/latest/files/PutRelativeFile.html"></a>.
+     */
+    public Object putRelativeFile() {
+        NuxeoPrincipal principal = (NuxeoPrincipal) session.getPrincipal();
+        String suggestedTarget = getHeader(SUGGESTED_TARGET, true);
+        if (suggestedTarget != null) {
+            suggestedTarget = Helpers.readUTF7String(suggestedTarget);
+        }
+        String relativeTarget = getHeader(RELATIVE_TARGET, true);
+        if (relativeTarget != null) {
+            relativeTarget = Helpers.readUTF7String(relativeTarget);
+        }
+
+        if (StringUtils.isBlank(suggestedTarget) && StringUtils.isBlank(relativeTarget)) {
+            throw new NotImplementedException();
+        }
+        if (StringUtils.isNotBlank(suggestedTarget) && StringUtils.isNotBlank(relativeTarget)) {
+            throw new NotImplementedException();
+        }
+
+        DocumentRef parentRef = doc.getParentRef();
+        if (!session.exists(parentRef) || !session.hasPermission(parentRef, SecurityConstants.ADD_CHILDREN)) {
+            throw new NotImplementedException();
```

Comment:
Not relevant from our point of view: for WOPI, the resource, ie. the document, is found.
And this case should never happen because [PutRelativeFile](https://wopi.readthedocs.io/projects/wopirest/en/latest/files/PutRelativeFile.html) cannot be called if `UserCanNotWriteRelative` is `true`.